scripts/mcp_rules_testing/test_phase5_digest.sh

```
@@ -0,0 +1,423 @@
+    sleep 1
+    DIGEST_COUNT_AFTER_1=$(exec_admin_silent "SELECT COUNT(*) FROM stats_mcp_query_digest WHERE tool_name = 'run_sql_readonly';")
+    log_verbose "Digest count after query 1: ${DIGEST_COUNT_AFTER_1}"
+    if [ "${DIGEST_COUNT_AFTER_1:-0}" -ge "${DIGEST_COUNT_BEFORE:-0}" ]; then
+        run_test "T5.16: Query tracked in digest (count: ${DIGEST_COUNT_BEFORE} -> ${DIGEST_COUNT_AFTER_1})" true
+    else
+        run_test "T5.16: Query tracked in digest" false
+    fi
+
+    # Test 5.17: Execute same query again and verify count_star increments
+    log_info "Executing same query again to test count_star increment..."
+    DIGEST_TEXT_CHECK=$(exec_admin_silent "SELECT digest_text FROM stats_mcp_query_digest WHERE tool_name = 'run_sql_readonly' AND digest_text LIKE '%test_phase5_table%' ORDER BY last_seen DESC LIMIT 1;")
```

Shellcheck (0.11.0)

[warning] DIGEST_TEXT_CHECK appears unused. Verify use (or export if used externally).

(SC2034)